filament/backend/src/metal/MetalEnums.h

```diff
@@ -25,6 +25,32 @@
 
 #include <Availability.h>
 
+namespace {
+    enum class TextureCompressionFamily {
+        EAC_ETC,
+        ASTC
+    };
+
+    bool IsSupported(id<MTLDevice> device, TextureCompressionFamily family) {
+        assert(device);
+#if defined(FILAMENT_IOS_SIMULATOR)
+        return false;
+#elif defined(IOS) && !TARGET_OS_MACCATALYST
+        return true;
```

Comment:
this is practically correct for Shapr3D supported devices, but would not be applicable for Filament on its own (e.g. upstreaming), as not all iOS devices support ASTC

Reply:
We refactored it with @veghbalazs to handle old devices: eedca31feddd656d676ff4fb7074527d20d6a612